src/Components/Admin/AssetModelViewer/assetModelViewer.js

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from "react";
+import "@google/model-viewer";
+import { makeStyles } from "@material-ui/core/styles";
+import Drawer from "@material-ui/core/Drawer";
+import CssBaseline from "@material-ui/core/CssBaseline";
+import Typography from "@material-ui/core/Typography";
+import Divider from "@material-ui/core/Divider";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import { getAsset, editAsset } from "../../../lib/assetEndpoints";
+import { useErrorHandler } from "react-error-boundary";
+import { jsonMock } from "./mockData.js";
+import SaveIcon from "@material-ui/icons/Save";
+
+const drawerWidth = 17;
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        display: "flex",
+    },
+    drawer: {
+        width: drawerWidth + "vw",
+        flexShrink: 0,
+    },
+    drawerPaper: {
+        width: drawerWidth + "vw",
+    },
+    // necessary for content to be below navbar
+    toolbar: theme.mixins.toolbar,
+    content: {
+        flexGrow: 1,
+        backgroundColor: theme.palette.background.default,
+        padding: theme.spacing(3),
+    },
+    textfields: {
+        "& .MuiTextField-root": {
+            margin: theme.spacing(2),
+            width: "15vw",
+        },
+    },
+}));
+
+export default function AssetModelViewer({
+    match: {
+        params: { assetId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [asset, setAsset] = useState({});
+    const [name, setName] = useState("");
+    const [objectType, setObjectType] = useState("");
+    const [buttonText, setButtonText] = useState("Save");
+    const [modelGLB, setModelGLB] = useState("");
+
+    useEffect(() => {
+        const getObjectAsset = async () => {
+            const data = await getAsset(assetId, handleError);
+            setName(data.name);
+            setObjectType(data.obj_type);
+            // TODO: Replace with asset's s3_link once it contains a legit link
```

Comment:
This currently contains the legit link right?
It should be `asset_prefix` + `data.s3_key`,
where `asset_prefix` == `http://localhost:8888/static/` in dev
and `asset_prefix` == `https://dcc-bp-public-dev.s3.us-east-2.amazonaws.com/` for deployment

Reply:
Updated PR with new environment variable